packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js

```diff
@@ -0,0 +1,163 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import * as d3 from 'd3'
+
+//TODO: allow styles to be configurable
+const VIEWER_STYLE = {
+  background: '#f8f8f8',
+  border: '#333',
+  border_size: '2',
+  data_point: '#3cc',
+}
+
+class LightCurveViewer extends React.Component {
+  
+  constructor() {
+    super()
+    
+    //React reference to the DOM.
+    //See: https://reactjs.org/docs/refs-and-the-dom.html
+    this.svgContainer = React.createRef()
+    
+    //D3 bits
+    this.d3svg = null
+    this.d3dataLayer = null
+  }
+  
+  componentDidMount () {
+    this.d3init()
+    this.d3loadData()
+  }
+
+  componentDidUpdate (prevProps) {
+    this.d3loadData()
+  }
+  
+  componentWillUnmount () {
+    this.d3exit()
+  }
+  
+  d3init () {
+    //Prepare the main SVG
+    //--------------------------------
+    this.d3svg = d3.select(this.svgContainer.current).append('svg')
+      .attr('class', 'light-curve-viewer')
+      //.attr('viewBox', `0 0 ${this.props.width} ${this.props.height}`)  //TODO: check
+      //.attr('style', 'width: 100%; height: 100%')  //TODO: check
+      .attr('width', this.props.width)
+      .attr('height', this.props.height)
+    
+    //Deco: Background layer
+    this.d3svg.append('rect')
+      .attr('width', this.props.width)
+      .attr('height', this.props.height)
+      .attr('fill', VIEWER_STYLE.background)
+      .attr('stroke', VIEWER_STYLE.border)
+      .attr('stroke-width', VIEWER_STYLE.border_size)
+    //--------------------------------
+    
+    //Prepare the data layer
+    //--------------------------------
+    this.d3dataLayer = this.d3svg.append('g')
+      .attr('class', 'data-layer')
+    //--------------------------------
+    
+    //Add interactive elements for panning and zooming
+    //--------------------------------
+    this.d3interfaceLayer = this.d3svg.append('rect')
+      .attr('class', 'interface-layer')
+      .attr('width', this.props.width)
+      .attr('height', this.props.height)
+      .style('fill', 'none')
+      .style('pointer-events', 'all')
+    
+    this.d3interfaceLayer.call(d3.zoom()
+      .scaleExtent([0.1, 10])
+      .on('zoom', () => { this.d3dataLayer.attr('transform', d3.event.transform) })
+    )
+    //--------------------------------
+  }
+  
+  d3exit () {
```

Comment:
I'd strongly suggest renaming this method - `exit` has its own meaning in the D3 data cycle (https://bost.ocks.org/mike/join/)

Reply:
Would you believe me when I tell you that I wasted a solid 15 minutes trawling through various dictionaries trying to figure out the antonym for "initialise" before settling for "exit", since it rhymed with "init"?

In any case, this is a good point; `d3exit()` might cause unnecessary confusion with`d3.selection.exit()` so will be renamed to `d3finish()`

Reply:
![screen shot 2018-10-29 at 17 00 30](https://user-images.githubusercontent.com/13952701/47666744-40012c80-db9c-11e8-8bd8-7449d3f66b92.png)

Go straight to hell, `synonym.com`